packages/flutter/lib/src/material/floating_action_button.dart

```diff
@@ -730,3 +711,148 @@ class _RenderChildOverflowBox extends RenderAligningShiftedBox {
     }
   }
 }
+
+// Generate a FloatingActionButtonThemeData that represents
+// the M2 default values. This was generated by hand from the
+// previous hand coded defaults for M2. It uses get method overrides
+// instead of properties to avoid computing values that we may not
+// need upfront.
+class _M2Defaults extends FloatingActionButtonThemeData {
+  _M2Defaults(BuildContext context, this.type, this.hasChild)
+      : theme = Theme.of(context),
+        colors = Theme.of(context).colorScheme;
+
+  final _FloatingActionButtonType type;
+  final bool hasChild;
+  final ThemeData theme;
+  final ColorScheme colors;
+
+  bool get isExtended => type == _FloatingActionButtonType.extended;
+  bool get isLarge => type == _FloatingActionButtonType.large;
+
+  @override Color? get foregroundColor => colors.onSecondary;
+  @override Color? get backgroundColor => colors.secondary;
+  @override Color? get focusColor => theme.focusColor;
+  @override Color? get hoverColor => theme.hoverColor;
+  @override Color? get splashColor => theme.splashColor;
+  @override double? get elevation => 6;
+  @override double? get focusElevation => 6;
+  @override double? get hoverElevation => 8;
+  @override double? get highlightElevation => 12;
+  @override ShapeBorder? get shape => isExtended ? const StadiumBorder() : const CircleBorder();
+  @override bool? get enableFeedback => true;
+  @override double? get iconSize => isLarge ? 36.0 : 24.0;
+
+  @override
+  BoxConstraints? get sizeConstraints => const BoxConstraints.tightFor(
+    width: 56.0,
+    height: 56.0,
+  );
+
+  @override
+  BoxConstraints? get smallSizeConstraints => const BoxConstraints.tightFor(
+    width: 40.0,
+    height: 40.0,
+  );
+
+  @override
+  BoxConstraints? get largeSizeConstraints => const BoxConstraints.tightFor(
+    width: 96.0,
+    height: 96.0,
+  );
+
+  @override
+  BoxConstraints? get extendedSizeConstraints => const BoxConstraints.tightFor(
+    height: 48.0,
+  );
+
+  @override double? get extendedIconLabelSpacing => 8.0;
+  @override EdgeInsetsGeometry? get extendedPadding => EdgeInsetsDirectional.only(start: hasChild && isExtended ? 16.0 : 20.0, end: 20.0);
+  @override TextStyle? get extendedTextStyle => theme.textTheme.button!.copyWith(letterSpacing: 1.2);
+}
+
+// BEGIN GENERATED TOKEN PROPERTIES
+
+// Generated code to the end of this file. Do not edit by hand.
+// These defaults are generated from the Material Design Token
+// database by the script dev/tools/gen_defaults/bin/gen_defaults.dart.
+
+// Generated version v0.74, 2022-01-06
+class _M3Defaults extends FloatingActionButtonThemeData {
+  _M3Defaults(this.context, this.type, this.hasChild)
+    : _colors = Theme.of(context).colorScheme,
+      _textTheme = Theme.of(context).textTheme;
+
+  final BuildContext context;
+  final _FloatingActionButtonType type;
+  final bool hasChild;
```

Comment:
Since its a private class, it probably doesn't matter, but should all these properties by private too?

Reply:
They are all constructor params, so I didn't want to make them private as I would have had to do the _param = param dance in the constructor and it didn't seem worth it for a private class.